addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,440 @@
+            }
+            /*
+             * ArrayList<String> polygon_lat_long_pairs = new ArrayList<String>();
+             * polygon_lat_long_pairs.add("0.00000,70.666011"); // -8.553029
+             * polygon_lat_long_pairs.add("37.934697,70.742227");
+             * polygon_lat_long_pairs.add("27.392184,52.542473");
+             * polygon_lat_long_pairs.add("2.250475,52.500440");
+             *
+             * if (!coordinate_is_inside_polygon(dlatitude, dlongitude, polygon_lat_long_pairs)) {
+             * updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+             * "Location not within SMHI Covered area.");
+             * configValid = false;
+             * }
+             */
+            if (configValid) {
+
+                apiRequest = String.format(URL, longitude, latitude);
+                updateStatus(ThingStatus.ONLINE);
+                startAutomaticRefresh(config.refresh);
+            }
+        });
+    }
+
+    private void startAutomaticRefresh(int refresh) {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                queryApiAndUpdateChannels();
+            };
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 10, refresh * 60, TimeUnit.SECONDS);
+        }
+    }
+
+    private void queryApiAndUpdateChannels() {
+        logger.debug("Running queryApiAndUpdateChannels");
+
+        String apiResponseJson = null;
+
+        SmhiData dataList = null;
+
+        try {
+            logger.debug("Quering SMHI API: " + apiRequest);
+
+            apiResponseJson = HttpUtil.executeUrl("GET", apiRequest, null, null, "application/json", SMHI_TIMEOUT);
+
+            dataList = gson.fromJson(apiResponseJson, SmhiData.class);
+
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime(new Date());
+            calendar.add(Calendar.HOUR_OF_DAY, 1);
+            calendar.set(Calendar.MINUTE, 0);
+            calendar.set(Calendar.SECOND, 0);
+            calendar.set(Calendar.MILLISECOND, 0);
+
+            Date currentvaliddate = calendar.getTime();
+
+            calendar.set(Calendar.HOUR_OF_DAY, 13);
+            calendar.add(Calendar.DATE, 1);
+            Date tomorrowvaliddate = calendar.getTime();
+
+            calendar.add(Calendar.DATE, 1);
+            Date day2validdate = calendar.getTime();
+
+            calendar.add(Calendar.DATE, 1);
+            Date day3validdate = calendar.getTime();
+
+            SimpleDateFormat cdtf = new SimpleDateFormat("yyyy-MM-dd HH:00:00");
+
+            SimpleDateFormat cdf = new SimpleDateFormat("yyyy-MM-dd");
+
+            final List<SmhiTimeSeries> timeSeries = dataList.getTimeSeries();
+
+            List<SmhiParameters> parameters;
+
+            HashMap<String, SmhiParameters> hashParameters = new HashMap<String, SmhiParameters>();
+
+            double temp;
+
+            double tmin, tmin1, tmin2, tmin3;
```

Comment:
It took me some time to find out what `tmin` and so on are used for. Please use more meaningul variable names (like `temperatureMinToday` or alike).

Reply:
Done